plugins/FrequencyShifter/FrequencyShifterEffect.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+ * FrequencyShifter.cpp
+ *
+ * Copyright (c) 2025 Lost Robot <r94231/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "FrequencyShifterEffect.h"
+
+#include "embed.h"
+#include "plugin_export.h"
+
+#include <algorithm>
+#include <cmath>
+#include <numbers>
+
+namespace lmms
+{
+
+extern "C"
+{
+Plugin::Descriptor PLUGIN_EXPORT frequencyshifter_plugin_descriptor =
+{
+	LMMS_STRINGIFY(PLUGIN_NAME),
+	"Frequency Shifter",
+	QT_TRANSLATE_NOOP("PluginBrowser", "A frequency shifter (not a pitch shifter) and barberpole phaser plugin"),
+	"Lost Robot <r94231/at/gmail/dot/com>",
+	0x0100,
+	Plugin::Type::Effect,
+	new PluginPixmapLoader("logo"),
+	nullptr,
+	nullptr,
+};
+}
+
+FrequencyShifterEffect::FrequencyShifterEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&frequencyshifter_plugin_descriptor, parent, key),
+	m_controls(this)
+{
+	m_ringBuf.resize(1, {0.f, 0.f});
+
+	connect(Engine::audioEngine(), &AudioEngine::sampleRateChanged,
+		this, &FrequencyShifterEffect::updateSampleRate);
+	updateSampleRate();
+}
+
+Effect::ProcessStatus FrequencyShifterEffect::processImpl(SampleFrame* buf, const fpp_t frames)
+{
+	const float twoPi = std::numbers::pi_v<float> * 2.0f;
```

Comment:
Might as well enforce compile-time evaluation of this constant:

```suggestion
	constexpr float twoPi = std::numbers::pi_v<float> * 2.0f;
```